lib/bolt/result_set.rb

```diff
@@ -0,0 +1,115 @@
+module Bolt; class ResultSet
+  attr_reader :results
+
+  include Enumerable
+
+  # We only want want to include these when puppet is loaded
+  def self.include_iterable
+    include(Puppet::Pops::Types::Iterable)
+    include(Puppet::Pops::Types::IteratorProducer)
+  end
+
+  def iterator
+    # TODO: does this work outside a compiler?
+    if Object.const_defined?(:Puppet) && Puppet.const_defined?(:Pops) &&
+        self.class.included_modules.include?(Puppet::Pops::Types::Iterable)
+      Puppet::Pops::Types::Iterable.on(@results, Bolt::Result)
+    else
+      @results.each
+    end
+  end
+
+  def initialize(results)
+    @results = results
+  end
+
+  def each
+    @results.each { |r| yield r }
+  end
+
+  def result_hash
+    @result_hash ||= @results.reduce({}) do |acc, result|
+      # TODO: What is the ID of a result target
+      acc[result.target.uri] = result
+      acc
+    end
+  end
+
+  def count
+    @results.size
+  end
+  alias length count
+  alias size count
+
+  def empty
+    @results.empty?
+  end
+  alias empty? empty
+
+  # Should this be nodes or targets?
+  def targets
+    results.map(&:target)
+  end
+
+  # Is this useful? For logging maybe?
+  def names
+    @results.map { |r| r.target.name }
+  end
+
+  def ok
+    @results.all? { |r| r.ok? }
+  end
+  alias ok? ok
+
+  def errorset
```

Comment:
This feels like a minor deviation from naming conventions. Would have expected `error_set` and `ok_set`.